exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet2/DrillParquetReader.java

```diff
@@ -123,14 +126,22 @@ private static MessageType getProjection(MessageType schema,
     Set<SchemaPath> selectedSchemaPaths = new LinkedHashSet<>();
 
     // get a list of modified columns which have the array elements removed from the schema path since parquet schema doesn't include array elements
+    // or if field is MAP then array/name segments are removed from the schema as well as obtaining elements by key is handled in EvaluationVisitor.
     List<SchemaPath> modifiedColumns = new LinkedList<>();
     for (SchemaPath path : columns) {
 
       List<String> segments = new ArrayList<>();
+      Type segmentType = schema;
       for (PathSegment seg = path.getRootSegment(); seg != null; seg = seg.getChild()) {
+
+        segmentType = getType(segmentType, seg);
```

Comment:
Please move down this and next line to be just before the ```if (isMap) {``` also add one line comment inside ```if``` describing why ```break;``` was called.